src/org/rust/cargo/project/CargoExternalSystemManager.kt

```diff
@@ -0,0 +1,93 @@
+package org.rust.cargo.project
+
+import com.google.gson.Gson
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.configurations.SimpleJavaParameters
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.externalSystem.ExternalSystemAutoImportAware
+import com.intellij.openapi.externalSystem.ExternalSystemManager
+import com.intellij.openapi.externalSystem.model.ProjectSystemId
+import com.intellij.openapi.externalSystem.service.project.ExternalSystemProjectResolver
+import com.intellij.openapi.externalSystem.service.project.autoimport.CachingExternalSystemAutoImportAware
+import com.intellij.openapi.externalSystem.task.ExternalSystemTaskManager
+import com.intellij.openapi.fileChooser.FileChooserDescriptor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Pair
+import com.intellij.util.Function
+import com.intellij.util.PathUtil
+import org.apache.commons.lang.StringUtils
+import org.rust.cargo.project.settings.*
+import org.rust.lang.config.RustConfigService
+
+import java.net.URL
+
+class CargoExternalSystemManager : ExternalSystemAutoImportAware,
+        ExternalSystemManager<CargoProjectSettings, CargoProjectSettingsListener, CargoSettings,
+                CargoLocalSettings, CargoExecutionSettings> {
+
+    override fun getSystemId(): ProjectSystemId {
+        return CargoProjectSystem.ID
+    }
+
+    override fun getSettingsProvider(): Function<Project, CargoSettings> {
+        return object : Function<Project, CargoSettings> {
+            override fun `fun`(project: Project): CargoSettings {
```

Comment:
I suggest to add util method and rewrite this functions using it.
```kotlin
fun <A, B> asIntellijFunction(function : (A) -> B) : Function<A, B> {
    return object : Function<A, B> {
        override fun `fun`(a: A): B {
            return function(a)
        }
    }
}
```
 This Functions looks crazy in Kotlin.